test/functional/p2p_tx_download.py

```diff
@@ -193,25 +203,90 @@ def test_notfound_fallback(self):
         peer_notfound.send_and_ping(msg_notfound(vec=[CInv(MSG_WTX, WTXID)]))  # Send notfound, so that fallback peer is selected
         peer_fallback.wait_until(lambda: peer_fallback.tx_getdata_count >= 1, timeout=1)
 
-    def test_preferred_inv(self, preferred=False):
-        if preferred:
-            self.log.info('Check invs from preferred peers are downloaded immediately')
+    def test_preferred_inv(self, connection_type: ConnectionType):
+        if connection_type == ConnectionType.WHITELIST:
```

Comment:
One more nit 😅 would a switch (match/case) work better here than if/else?

Reply:
a bit too nitty, keeping as is :)